base/stat.jl

```diff
@@ -301,7 +301,7 @@ function filemode_string(mode)
         end
         complete && write(str, "-")
     end
-    return String(take!(str))
+    return unsafe_takestring!(str)
```

Comment:
Why is this guaranteed to be safe?

Reply:
Similar here, we can guarantee no reference to `str` is leaked out of the function.